src/main/java/org/opentripplanner/model/Timetable.java

```diff
@@ -432,12 +430,12 @@ public boolean isValidFor(LocalDate serviceDate) {
   // TODO maybe put this is a more appropriate place
   public void setServiceCodes(Map<FeedScopedId, Integer> serviceCodes) {
     for (TripTimes tt : this.tripTimes) {
-      tt.setServiceCode(serviceCodes.get(tt.getTrip().getServiceId()));
+      ((RealTimeTripTimes) tt).setServiceCode(serviceCodes.get(tt.getTrip().getServiceId()));
```

Comment:
Is this cast going to stay or will it be refactored in a later PR?

Reply:
I think we should be able to get rid of it.
- In general the updaters should work on RealTimeTripTimes and the API they use should provide them. There is several ways to achieve this - one way is to keep all RT stuff in separate data structures. Then when searching for "a" TripTime we can have a facade witch delegate to both planned and realtime - but there are other options.
- I think we are getting rid of the service-code (keeping serviceIds for reference). But, I want another calendar implementation witch works better with Raptor.